### PR TITLE
"How Optimism Works" section

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Adds the How Optimism Works page. 

Also makes a couple of updates to protocol specs section to add roadmap section

### CODE REVIEW

**Comment by smartcontracts** [comment]
I tried to cut this down to the absolute basics required to understand how Optimism works. Is it too simplistic? Are there things that need to be expanded?

File: src/docs/how-optimism-works/README.md [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+This significantly reduces overall transaction fees by spreading fixed costs over all of the transactions in a given batch.
+Block producers also apply some basic compression techniques to minimize the amount of data published to Ethereum.
+Only one block producer is active at any given time.
+For the moment, [Optimism PBC](https://www.optimism.io/) runs the only block producer.
+Refer to the below section about [Sequencer decentralization](#sequencer-decentralization) for more information.
+
+![op loves eth](../../assets/docs/how-optimism-works/3.png)
+
+Alternatively, users can skip block producers entirely and submit their transactions directly to the `CanonicalTransactionChain`.
+This is typically more expensive but has the advantage of being resistant to censorship by block producers.
```
Comment (smartcontracts) [inline src/docs/how-optimism-works/README.md:118]: Add a sentence here about how this is the way that L1 contracts can send data to L2.

File: src/docs/how-optimism-works/README.md [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+(diagram)
+
+### Withdrawals
+
+The process of sending data from Optimism back to Ethereum is somewhat more involved.
+In this direction, we need to be able to make provable statements about the state of Optimism to contracts sitting on Ethereum.
+
+Making provable statements about the state of Optimism requires a [cryptographic commitment](https://en.wikipedia.org/wiki/Commitment_scheme) in the form of the root of the Optimism's [state trie](https://medium.com/@eiki1212/ethereum-state-trie-architecture-explained-a30237009d4e).
+Optimism's state is updated after each block, so this commitment will also change after every block.
+Commitments are regularly published to smart contract on Ethereum called the [`StateCommitmentChain`](https://etherscan.io/address/0xBe5dAb4A2e9cd0F27300dB4aB94BeE3A233AEB19).
```
Comment (smartcontracts) [inline src/docs/how-optimism-works/README.md:157]:
```suggestion
Commitments are regularly published to a smart contract on Ethereum called the [`StateCommitmentChain`](https://etherscan.io/address/0xBe5dAb4A2e9cd0F27300dB4aB94BeE3A233AEB19).
```

File: src/docs/how-optimism-works/README.md [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+
+Optimism is an "Optimistic Rollup," which is basically just a fancy way of describing a blockchain that piggy-backs off of the security of another "parent" blockchain.
+Specifically, Optimistic Rollups take advantage of the consensus mechanism (like PoW or PoS) of their parent chain instead of providing their own.
+In Optimism's case this parent blockchain is Ethereum.
+
+![op loves eth](../../assets/docs/how-optimism-works/1.png)
+
+### Block storage
+
+Optimism blocks are published by sending a transaction to a special smart contract on Ethereum called the [`CanonicalTransactionChain`](https://etherscan.io/address/0x5E4e65926BA27467555EB562121fac00D24E9dD2) (or CTC for short).
```
Comment (smartcontracts) [inline src/docs/how-optimism-works/README.md:96]: Add a few sentences about how Optimism blocks are identical in structure to Ethereum blocks

File: src/docs/how-optimism-works/README.md [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+(image)
+
+Optimism nodes are made up of two primary components, the Ethereum data indexer and the Optimism client software.
+The Ethereum data indexer, also called the ["data transport layer"](https://github.com/ethereum-optimism/optimism/tree/develop/packages/data-transport-layer) (or DTL), pieces together the Optimism blockchain from blocks published to the `CanonicalTransactionChain` contract.
+
+(image)
+
+Once a block is indexed, the Optimism client software will pull this block in and execute it.
+Optimism's client software is an *almost* completely vanilla version of [Geth](https://github.com/ethereum/go-ethereum), which means Optimism is close to identical to Ethereum under the hood.
+We refer to this architecture as ["EVM Equivalence"](https://medium.com/ethereum-optimism/introducing-evm-equivalence-5c2021deb306).
```
Comment (smartcontracts) [inline src/docs/how-optimism-works/README.md:138]: Maybe add a sentence here about why we chose to do this and how it relates to our design principles

File: src/docs/how-optimism-works/README.md [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+
+(insert node diagram)
+
+### Deposits
+
+Since users can create blocks on Optimism by triggering the `CanonicalTransactionChain` on Ethereum, it's very easy to send data from Ethereum to Optimism.
+User-created blocks can include arbitrary transaction data and will appear to originate from the address that generated the block.
```
Comment (smartcontracts) [inline src/docs/how-optimism-works/README.md:145]:
```suggestion
User-created blocks can include transactions that will appear to originate from the address that generated the block.
```

File: src/docs/how-optimism-works/README.md [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+Contracts on Ethereum can use this feature to, for example, deposit some assets from Ethereum into Optimism.
```
Comment (smartcontracts) [inline src/docs/how-optimism-works/README.md:146]: Can consider expanding on how this deposit would actually work

File: src/docs/how-optimism-works/README.md [hunk 7]
```diff
@@ -0,0 +1,395 @@
...
+
+(diagram)
+
+### Withdrawals
+
+The process of sending data from Optimism back to Ethereum is somewhat more involved.
+In this direction, we need to be able to make provable statements about the state of Optimism to contracts sitting on Ethereum.
```
Comment (smartcontracts) [inline src/docs/how-optimism-works/README.md:153]: > we need to be able to make provable statements about the state of Optimism

Should be explained better

File: src/docs/how-optimism-works/README.md [hunk 8]
```diff
@@ -0,0 +1,395 @@
...
+### Sequencer decentralization
+
+Optimism PBC currently runs the only Optimism Sequencer node.
+We're working hard to completely decentralize the Sequencer selection process so that anyone can participate in the network as a block producer.
+Getting to full Sequencer decentralization is a challenging process that requires careful consideration.
+In particular, we need to manage the impact of [Miner Extractable Value](https://ethereum.org/en/developers/docs/mev/) that Sequencer nodes have access to.
+
+TODO: expand this section
+
+<!-- ## Why Optimistic Rollups?
```
Comment (smartcontracts) [inline src/docs/how-optimism-works/README.md:217]: Everything below here is just scrap and stuff cut from previous iterations of this page

File: src/docs/how-optimism-works/README.md [hunk 9]
```diff
@@ -0,0 +1,395 @@
...
+
+(image)
+
+Once a block is indexed, the Optimism client software will pull this block in and execute it.
+Optimism's client software is an *almost* completely vanilla version of [Geth](https://github.com/ethereum/go-ethereum), which means Optimism is close to identical to Ethereum under the hood.
+We refer to this architecture as ["EVM Equivalence"](https://medium.com/ethereum-optimism/introducing-evm-equivalence-5c2021deb306).
+
+(insert node diagram)
+
+### Deposits
```
Comment (elenadimitrova) [inline src/docs/how-optimism-works/README.md:142]: Shouldn't the "Deposits" and Withdrawals" sections be renamed to something more generic e.g. "Transferring assets and data between the layers"?

File: src/docs/how-optimism-works/README.md [hunk 10]
```diff
@@ -0,0 +1,395 @@
...
+
+(diagram)
+
+This process is guaranteed to always correctly determine whether or not the published state root is valid.
+Bad state roots will always be invalidated as long as at least one user is willing to perform this challenge process.
+Under the assumption that at least one such user exists, we can be certain that any state root that has passed its 7 day waiting period without being invalidated by a challenge must be valid.
+Once we can be sure that a state root is valid, users can begin to use that state root to create Merkle proofs about the current state of Optimism.
+
+## Roadmap
+
```
Comment (elenadimitrova) [inline src/docs/how-optimism-works/README.md:193]: We should link to the website roadmap here https://www.optimism.io/about

File: src/docs/how-optimism-works/README.md [hunk 11]
```diff
@@ -0,0 +1,395 @@
...
+### Next gen fault proofs
```
Comment (elenadimitrova) [inline src/docs/how-optimism-works/README.md:194]: We added a page about the Fault proofs here https://github.com/ethereum-optimism/community-hub/pull/326/files#diff-81969a92e6f452745c3e18d617323d2ea3c895b79e86a48d813b8e3f8ce64ac5  there is also a dedicated page about the sequencer in the protocol section of docs. I suggest we reference the website for high lever easy-to-digest roadmap reference and the protocol sections for more detailed references without reinventing this content.

File: src/docs/how-optimism-works/README.md [hunk 12]
```diff
@@ -0,0 +1,395 @@
...
+
+## Introduction
+
+Hello!
+By the time you've finished reading this page you should have a foundational understanding of how Optimism makes Ethereum transactions cheaper and faster, the approach that Optimism is taking to scaling both Ethereum and Ethereum's values, and why Optimism is the best place to build your next Ethereum-native app.
+
+We've tried to make this guide as comprehensive as possible while still keeping the content accessible to most readers.
+Some content on this page is geared towards readers with a technical background but should still be legible to those with a basic understanding of how blockchains work.
+Generally speaking, we err on the side of simplicity and approachability.
+Readers interested in a low-level look into Optimism under the hood should refer to the [Protocol](http://localhost:8080/docs/protocol/) section of this website.
```
[the remark this answers is not in the document]
Comment (Nicca42) [reply src/docs/how-optimism-works/README.md:16]: You can do relative pathing with markdown. Should be something like `../protocol/README.md`

File: src/docs/how-optimism-works/README.md [hunk 13]
```diff
@@ -0,0 +1,395 @@
...
+
+Optimism, like Ethereum, is a network of nodes.
+Ethereum nodes download blocks from Ethereum's p2p network.
+Optimism nodes instead download blocks directly from the `CanonicalTransactionChain` contract.
+These blocks are then executed deterministically, just like any other blockchain.
+
+(image)
+
+Optimism nodes are made up of two primary components, the Ethereum data indexer and the Optimism client software.
+The Ethereum data indexer, also called the ["data transport layer"](https://github.com/ethereum-optimism/optimism/tree/develop/packages/data-transport-layer) (or DTL), pieces together the Optimism blockchain from blocks published to the `CanonicalTransactionChain` contract.
```
Comment (Nicca42) [inline src/docs/how-optimism-works/README.md:132]: Can we update this to "the ["Data Transport Layer (DTL)](..."

Comment (Nicca42) [reply src/docs/how-optimism-works/README.md:132]: maybe also "reconstructs the Optimism blockchain..."

File: src/docs/how-optimism-works/README.md [hunk 14]
```diff
@@ -0,0 +1,395 @@
...
+
+(image)
+
+Once a block is indexed, the Optimism client software will pull this block in and execute it.
```
Comment (Nicca42) [inline src/docs/how-optimism-works/README.md:136]: This sentence is not clear. 

Once a block is indexed by the DTL, the Optimism client software will take the indexed block and ???

 I am not sure I even understand this well enough to rephrase it. What is it executing? The already created and stored block from the DTL? And why is it executing it? To reconstruct state? To verify execution? It is not clear.

File: src/docs/how-optimism-works/README.md [hunk 15]
```diff
@@ -0,0 +1,395 @@
...
+Before any code was written, Optimism was just a goal: *make Ethereum mainstream*.
+
+We just needed to figure out exactly how to turn that lofty goal into a reality.
+Constructing a new L1 system was out of the question.
+Optimism was designed to scale Ethereum, not create a new chain that competes with it.
+This left only two serious options: Optimistic Rollups or ZK Rollups.
+
+### Simple
+
+ZK Rollups are extremely cool.
```
Comment (Nicca42) [inline src/docs/how-optimism-works/README.md:228]: I have been asked about this a few times, not sure if this is the best spot for this but I would love to have a section about ZK rollups and our feelings towards them as they figure themselves out

File: src/docs/how-optimism-works/README.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (smartcontracts) [reply src/docs/how-optimism-works/README.md:16]: Oops!

File: src/docs/how-optimism-works/README.md (hunk 9 above)
Comment (smartcontracts) [reply src/docs/how-optimism-works/README.md:163]: Yeah I think that's a good idea

File: src/docs/how-optimism-works/README.md [hunk 16]
```diff
@@ -0,0 +1,436 @@
...
+Optimism is therefore secure against large block reorganizations as long as the Ethereum consensus mechanism is too.
+It's through this relationship (in part, at least) that Optimism derives its security properties from Ethereum.
+
+<div align="center">
+<img width="400" src="../../assets/docs/how-optimism-works/2.png">
+</div>
+
+### Block production
+
+Users can have their transactions added to the Optimism chain in one of two ways: by sending the transaction to a block producer (which we call a "Sequencer") or by submitting their transactions directly to the `CanonicalTransactionChain`.
```
Comment (maurelian) [inline src/docs/how-optimism-works/README.md:117]: I think it's worth clarifying why it makes sense to call the "Block Producer" a "Sequencer". ie. it's not immediately obvious that a major part of producing a block is choosing which tx to include and in what order.

File: src/docs/how-optimism-works/README.md [hunk 17]
```diff
@@ -0,0 +1,436 @@
...
+
+Optimism is an "Optimistic Rollup," which is basically just a fancy way of describing a blockchain that piggy-backs off of the security of another "parent" blockchain.
+Specifically, Optimistic Rollups take advantage of the consensus mechanism (like PoW or PoS) of their parent chain instead of providing their own.
+In Optimism's case this parent blockchain is Ethereum.
+
+<div align="center">
+<img width="400" src="../../assets/docs/how-optimism-works/1.png">
+</div>
+
+### Block storage
```
Comment (maurelian) [inline src/docs/how-optimism-works/README.md:96]: Similar to the comment made just below here, I see a need to clarify the relationship between blocks and transaction. TBH I myself don't typically think of transaction batches as blocks, though I can see now how that makes sense since EVM equivalence.

File: src/docs/how-optimism-works/README.md [hunk 18]
```diff
@@ -0,0 +1,436 @@
...
+We refer to this architecture as ["EVM Equivalence"](https://medium.com/ethereum-optimism/introducing-evm-equivalence-5c2021deb306) and it means that most Ethereum tools (even the most complex ones) "just work" with Optimism.
+
+The Optimism client software continuously monitors the DTL for newly indexed blocks.
+When a new block is indexed, the client software will download it and execute the transactions included within it.
+The process of executing a transaction on Optimism is the same as on Ethereum: we load the Optimism state, apply the transaction against that state, and then record the resulting state changes.
+This process is then repeated for each new block indexed by the DTL.
+
+(insert node diagram)
+
+### Moving assets between layers
```
Comment (maurelian) [inline src/docs/how-optimism-works/README.md:164]: I think we should address address aliasing here. 
In particular it's useful to callout that the aliasing and de-aliasing occur in the contract layer, and geth just sets tx.origin/msg.sender based on logs emitted by the CTC.

File: src/docs/how-optimism-works/README.md [hunk 19]
```diff
@@ -0,0 +1,436 @@
...
+This process is guaranteed to always correctly determine whether or not the published state root is valid.
+Bad state roots will always be invalidated as long as at least one user is willing to perform this challenge process.
+Under the assumption that at least one such user exists, we can be certain that any state root that has passed its 7 day waiting period without being invalidated by a challenge must be valid.
+Once we can be sure that a state root is valid, users can begin to use that state root to create Merkle proofs about the current state of Optimism.
+
+## Roadmap
+
+### Next gen fault proofs
+
+On November 11th, 2021, the Optimism protocol went through its biggest upgrade to date.
```
Comment (maurelian) [inline src/docs/how-optimism-works/README.md:237]: @elenadimitrova I believe this is the content you moved into #329 right?

File: src/docs/how-optimism-works/README.md (hunk 15 above)
Comment (elenadimitrova) [reply src/docs/how-optimism-works/README.md:203]: I have added an FAQs section under "How Optimism works" where this might fit better.

File: src/docs/how-optimism-works/README.md (hunk 19 above)
Comment (elenadimitrova) [reply src/docs/how-optimism-works/README.md]: I have moved the roadmap details from here to the Protocol specs holding page.

File: src/docs/how-optimism-works/README.md [hunk 20]
```diff
@@ -0,0 +1,370 @@
...
+Specifically, Optimistic Rollups take advantage of the consensus mechanism (like PoW or PoS) of their parent chain instead of providing their own.
+In Optimism's case this parent blockchain is Ethereum.
+
+<div align="center">
+<img width="400" src="../../assets/docs/how-optimism-works/1.png">
+</div>
+
+### Fault proofs
+
+In Optimistic rollups state changes are considered pending for a period of time, called the "challenge window".  If a proposed state change goes unchallenged for the duration of the challenge window (currently set to 7 days), then it is considered final.
```
Comment (smartcontracts) [inline src/docs/how-optimism-works/README.md:98]: I think the framing here may reinforce the misconception that transactions/state can be rolled back. I'm going to see how I can rephrase it to avoid this.

File: src/docs/how-optimism-works/README.md (hunk 16 above)
Comment (elenadimitrova) [reply src/docs/how-optimism-works/README.md]: I pulled some of the sequencer information you already had in the protocol specs to clarify this see this paragraph https://github.com/ethereum-optimism/community-hub/pull/325/files#diff-a2377cc33b3952a9fb6ec3d989f68cb9c0f1334fdc6956ec290d634c8d6c82a2R117

**Review by Nicca42 — APPROVED** [review]
